src/reanimated2/shareables.ts

```diff
@@ -241,14 +241,28 @@ Offending code was: \`${getWorkletCode(value)}\``);
         shareableMappingCache.set(value, inaccessibleObject);
         return inaccessibleObject;
       }
-      if (__DEV__) {
-        // we freeze objects that are transformed to shareable. This should help
-        // detect issues when someone modifies data after it's been converted to
-        // shareable. Meaning that they may be doing a faulty assumption in their
-        // code expecting that the updates are going to automatically populate to
+      if (__DEV__ && !isHostObject(value) && isPlainJSObject(value)) {
```

Comment:
Would it make sense to move `&& !isHostObject(value)` inside `isPlainJSObject`?

Reply:
I moved this part to a separate function in 9d40d1f.